### PR TITLE
ibi mgmt: add siteconfig option to ibio test

### PR DESCRIPTION
**Pull request by trewest**
- Add options for deploying with siteconfig to IBIO tests
- Add test case for IPv6 proxy + DHCP

### CODE REVIEW

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 1]
```diff
@@ -79,11 +87,34 @@ var _ = Describe(
...
 			})
 
+		It("through siteconfig operator is successful in an IPv6 proxy-enabled environment with DHCP networking",
+			reportxml.ID("76642"), func() {
+				if MGMTConfig.StaticNetworking {
+					Skip("Cluster must use DHCP networking")
+				}
+
+				if !MGMTConfig.SiteConfig {
+					Skip("Cluster must be deployed through siteconfig operator")
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:106]:
```suggestion
					Skip("Cluster was not opted to be deployed through siteconfig operator")
```

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 2]
```diff
@@ -305,6 +336,128 @@ func createIBIOResouces(addressFamily string) {
...
+		}
+
+		By("Add node entry for " + host)
+
+		siteconfigNode := siteconfig.NewNodeBuilder(host, bmcAddress, info.BMC.MACAddress, host, ibiNodeTemplateName,
+			nodeTemplateConfigmap[0].Object.Namespace).WithAutomatedCleaningMode("disabled")
+
+		if MGMTConfig.StaticNetworking {
+			if len(info.Network.Interfaces) != 1 {
+				Skip("Cannot support nodes with more than one network interface")
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:410]: why not?

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go (hunk 1 above)
Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:106]: How is it different?

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go (hunk 2 above)
Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:410]: Just not something we have implemented yet

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go (hunk 1 above)
Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:106]: It must be deployed via siteconfig for this test to run

Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:106]: IMO it makes sense given the context of the name of the test - `through siteconfig operator is successful in an IPv6 proxy-enabled environment with DHCP networking`.

i.e. what is being tested here is performing an image based installation through siteconfig operator.

 `Skip("Cluster must be deployed through siteconfig operator")` is just defining the constraint that the test must be run with the `ECO_LCA_IBI_SITECONFIG` value set to true. 

So I'm assuming you have the same feeling about `Skip("Cluster must use DHCP networking")`?

Comment (achuzhoy) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:106]: yes (it's how someone looking at our tests from outside the team will perceive the message with an untrained eye)
And extra hard feelings about siteconfig operator - after realizing how confusing it is for those outside of the team.

Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:106]: I think the clearest I can make this message is something like `Skip("ECO_LCA_IBI_SITECONFIG must be set to true")`. From my POV, I don't see how someone from the outside can glean any additional information from "Cluster was not opted to be deployed through siteconfig operator" vs "Cluster must be deployed through siteconfig operator"

Comment (achuzhoy) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:106]: "Cluster must be deployed through siteconfig operator" seems like instruction to always follow to me

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**